bot/cogs/moderation/silence.py

```diff
@@ -0,0 +1,199 @@
+import asyncio
+import datetime
+import logging
+from collections import namedtuple
+from typing import Optional
+
+from discord import TextChannel
+from discord.ext import commands
+from discord.ext.commands import Context
+
+from bot.bot import Bot
+from bot.cogs.moderation.modlog import ModLog
+from bot.constants import (STAFF_ROLES, Channels, Colours, Emojis, Guild,
+                           Icons, Roles)
+from bot.converters import SilenceDurationConverter
+from bot.utils import time
+from bot.utils.checks import with_role_check
+from bot.utils.scheduling import Scheduler
+
+log = logging.getLogger(__name__)
+
+SilencedChannel = namedtuple(
+    "SilencedChannel", ("id", "ctx", "silence", "stop"))
+
+
+class UnsilenceScheduler(Scheduler):
+    """Scheduler for unsilencing channels"""
+
+    def __init__(self, bot: Bot):
+        super().__init__()
+
+        self.bot = bot
+
+    async def schedule_unsilence(self, channel: SilencedChannel) -> None:
+        """Schedule expiration for silenced channels"""
+        await self.bot.wait_until_guild_available()
+
+        log.debug("Scheduling unsilencer")
+
+        self.schedule_task(channel.id, channel)
```

Comment:
There is no real reason to put spaces in here
```suggestion
        await self.bot.wait_until_guild_available()
        log.debug("Scheduling unsilencer")
        self.schedule_task(channel.id, channel)
```